File: kolokwia/Zadania_powt/Z3/student.py
from osoba import Osoba

class Student(Osoba):
    _rok: int
    _nr_gr: int
    _nr_albumu: int

    def __init__(self, rok: int, nr_gr: int, nr_albumu: int, imie: str, nazwisko: str, rok_urodzenia: int) -> None:
        super(Student, self).__init__(imie, nazwisko, rok_urodzenia)
        self._rok = rok
        self._nr_gr = nr_gr
        self._nr_albumu = nr_albumu
        

File: Java/Egzamin/Eg1/src/Sosnowski/powtorka/J1/Student.java
package Sosnowski.powtorka.J1;

import java.time.LocalDate;

public class Student extends Osoba implements Cloneable, Comparable<Osoba>{

    private double sredniaOcen;

    public Student(String nazwisko, LocalDate dataUrodzenia, double sredniaOcen) {
        super(nazwisko, dataUrodzenia);
        this.sredniaOcen = sredniaOcen;
    }

    public double getSredniaOcen() {
        return sredniaOcen;
    }

    public void setSredniaOcen(double sredniaOcen) {
        this.sredniaOcen = sredniaOcen;
    }

    @Override
    public Student clone() {
        Student clone = (Student) super.clone();
        return clone;
    }

    @Override
    public int compareTo(Osoba o) {
        int result = super.compareTo(o);
        if (result == 0 && o instanceof Student) {
            Student s = (Student) o;
            return Double.compare(sredniaOcen, s.sredniaOcen);
        }
        return result;
    }
}
